vcs.mk

```diff
@@ -11,34 +11,56 @@ endif
 CLOCK_PERIOD ?= 1.0
 RESET_DELAY ?= 777.7
 
+#----------------------------------------------------------------------------------------
+# gcc configuration/optimization
+#----------------------------------------------------------------------------------------
+# -flto slows down compilation on small-memory and breaks on firesim-manager
+CMODE := -O3 -fbranch-probabilities -march=native
+
+VCS_CXXFLAGS = \
+	$(CXXFLAGS) \
+	$(CMODE) \
+	-I$(RISCV)/include \
+	-I$(dramsim_dir) \
+	-std=c++11 \
+	$(EXTRA_SIM_CXXFLAGS)
+
+VCS_LDFLAGS =	\
+	$(LDFLAGS) \
+	-L$(RISCV)/lib \
+	-Wl,-rpath,$(RISCV)/lib \
+	-L$(sim_dir) \
+	-L$(dramsim_dir) \
+	-lfesvr \
+	-ldramsim \
+	$(EXTRA_SIM_LDFLAGS)
+
+# vcs requires LDFLAGS to not include library names (i.e. -l needs to be separate)
 VCS_CC_OPTS = \
-	-CC "-I$(RISCV)/include" \
-	-CC "-I$(dramsim_dir)" \
-	-CC "-std=c++11" \
-	-CC "$(EXTRA_SIM_CC_FLAGS)"
+	-CFLAGS "$(VCS_CXXFLAGS)" \
+	-LDFLAGS "$(filter-out -l%,$(VCS_LDFLAGS))" \
+	$(filter -l%,$(VCS_LDFLAGS))
 
 VCS_NONCC_OPTS = \
-	$(dramsim_lib) \
```

Comment:
Where did these two flags go?

Reply:
They were moved into the LDFLAGs section (see the `-L` and `-l` flags)

Reply:
Ah I didn't realize we wouldn't need the full path anymore, sorry.